### PR TITLE
Added parsing of datetime when using newtonsoft.json

### PR DESCRIPTION
**Pull request by pnobre**
When using Newtonsoft.Json and trying to parse an object containing a date time, it would fail with an exception "Invalid JToken type".

This PR fixes that behavior.

### CODE REVIEW

File: test/Tests/Tests.fs [hunk 1]
```diff
@@ -555,7 +564,7 @@ let tests = [
             //yield testProperty "float32" (roundtrip<float32>)  // not handled by FsCheck
             yield testProperty "string" (roundtrip<string>)
             yield testProperty "decimal" (roundtrip<decimal>)
-            yield testProperty "DateTime" (roundtrip<DateTime>)
+            //yield testProperty "DateTime" (roundtrip<DateTime>)
```
Comment (wallymathieu) [inline test/Tests/Tests.fs:567]: shouldn't this be included?

Comment (pnobre) [reply test/Tests/Tests.fs]: Yes. Sorry, forgot it commented... already addressed.

File: src/Fleece/Fleece.fs [hunk 2]
```diff
@@ -333,6 +336,7 @@ module SystemTextJson =
         | String = 4
         | Bool   = 5
         | Null   = 6
+        | Date   = 7
```
Comment (wallymathieu) [inline src/Fleece/Fleece.fs:339]: perhaps:
```f#
#if NEWTONSOFT
```
?

Comment (pnobre) [reply src/Fleece/Fleece.fs:340]: Makes sense. Addressed.

**Pull request merged**